src/client/address_book/impl.cc

```diff
@@ -444,24 +444,26 @@ std::unique_ptr<const kovri::core::IdentHash> AddressBook::GetLoadedAddressIdent
 // Used only by HTTP Proxy
 void AddressBook::InsertAddressIntoStorage(
     const std::string& address,
-    const std::string& base64) {
-  kovri::core::IdentityEx ident;
+    const std::string& base64)
+{
   try
     {
+      kovri::core::IdentityEx ident;
       ident.FromBase64(base64);
+      if (!m_Storage)
+        m_Storage = GetNewStorageInstance();
+      m_Storage->AddAddress(ident);
+      auto const ident_hash = ident.GetIdentHash();
```

Comment:
Please use appropriate qualifier `auto& const`

Reply:
Not being a smartass, but the [C++ ISO](https://isocpp.org/wiki/faq/const-correctness#const-ref-nonsense) site calls this pattern nonsense:
```
 But that is redundant — references are always const ... “X& const x” is functionally equivalent to “X& x”
```

So would you like me to remove the const qualifier, and add the reference?

Reply:
>references are always const

Yes, but this is taken out of context when using `auto` as an lvalue. Case in point:

```cpp
#include <iostream>

class Test
{
 public:
  Test() : m_test("test") {}

  const std::string& func() const
  {
    return m_test;
  }

 private:
  std::string m_test;
};

int main()
{
  Test t;

  std::cout << t.func() << std::endl;
  //const auto a = t.func();  // Fortunately, won't compile
  //auto const a = t.func();  // Fortunately, won't compile
  //auto& a = t.func();  // Fortunately, won't compile
  //const auto& a = t.func();  // Fortunately, won't compile
  //auto& const a = t.func();  // Fortunately, won't compile
  auto a = t.func();  // Unfortunately, will compile

  a = "uh oh";
  std::cout << a << std::endl;

  return 0;
}
```

When one uses `auto`, one mustn't forget the modifiers when needed. As for the committee, they have a mostly valid point, but:

- For this review, ensuring that the modifier is in place was the biggest point
- If the API wasn't properly made (i.e., `GetIdentHash` did *not* return const reference), then `error: ‘const’ qualifiers cannot be applied to ‘auto&’` would fortunately arise when attempting to build with `auto& const` - thus quickly pointing out the API issue. The same couldn't be said with `auto const` or `const auto` (which is what you originally had) or `const auto&`.

But, of course, this is all arguable so, whatever works should suffice (just don't alter the member or make a copy).